### PR TITLE
[Messenger] Added SQS Messenger docs

### LINKED ISSUE #13093: [Messenger] Add SQS transport
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#32454
| PR author(s) | @jderusse
| Merged in    | 5.1-dev

### PR DESCRIPTION
**Pull request by Nyholm**
This will close #13328 and fix #13093

### CODE REVIEW

**Comment by Nyholm** [comment]
This PR should target 5.1

**Comment by Nyholm** [comment]
There you go @OskarStark. Im done with this PR now

File: messenger.rst [hunk 1]
```diff
@@ -1268,6 +1268,72 @@ during a request::
...
+    # .env
+    MESSENGER_TRANSPORT_DSN=sqs://AKIAIOSFODNN7EXAMPLE:j17M97ffSVoKI0briFoo9a@sqs.eu-west-3.amazonaws.com/messages
+    #MESSENGER_TRANSPORT_DSN=sqs://localhost:9494/messages?sslmode=disable
+
+.. note::
+
+    The transport will automatically create queues that are needed. This
+    can be disabled setting the "auto_setup" option to ``false``.
+
+A numbeXr of options can be configured via the DSN or via the ``options`` key
```
Comment (jderusse) [inline messenger.rst]: `number`

**Review by jderusse — APPROVED** [review]
Thank you for taking care of this PR @Nyholm

File: messenger.rst (hunk 1 above)
Comment (Nyholm) [reply messenger.rst]: Ah.. I was testing if the docker build script automatically updated.. :)

**Comment by jderusse** [comment]
should we talk about async-aws, and authentication mechanism that fallback to .ini files, EC2/Lambda roles, Eks webidentity... and key/secret are not required

**Comment by Nyholm** [comment]
Hm.. Im not sure we should. Access key/secret is a really good way for new users. Advanced users will look at the code and find AsyncAws and find their way...  
We dont want to duplicate the docs from AsyncAws

File: messenger.rst [hunk 2]
```diff
@@ -1268,6 +1268,72 @@ during a request::
         :class:`Symfony\\Bundle\\FrameworkBundle\\Test\\KernelTestCase`
         or :class:`Symfony\\Bundle\\FrameworkBundle\\Test\\WebTestCase`.
 
+Amazon SQS
+~~~~~~~~~~
+
+.. versionadded:: 5.1
+
+    The Amazon SQS transport has been added in Symfony 5.1
```
Comment (javiereguiluz) [inline messenger.rst:1159]: We'll make this change while merging, but for your information, the `versionadded` messages are standardized as: `XXX was introduced in Symfony N.N.`

File: messenger.rst [hunk 3]
```diff
@@ -1268,6 +1268,72 @@ during a request::
...
+                    seconds
+visibility_timeout  Amount of seconds the message will      Queue's configuration
+                    not be visible (`Visibility Timeout`_)
+auto_setup          Whether the table should be created     true
+                    automatically during send / get.
+==================  ======================================  ======================
+
+.. note::
+
+    The ``wait_time`` parameter define the maximum duration Amazon SQS should
```
Comment (javiereguiluz) [inline messenger.rst]: `define` -> `defines`

File: messenger.rst [hunk 4]
```diff
@@ -1268,6 +1268,72 @@ during a request::
...
+    wait until a message is available in a queue before sending a response.
+    It helps reducing the cost of using Amazon SQS by eliminating the number
+    of empty responses.
+
+    The ``poll_timeout`` parameter define the duration the receiver should wait
```
Comment (javiereguiluz) [inline messenger.rst]: `define` -> `defines`

File: messenger.rst [hunk 5]
```diff
@@ -1268,6 +1268,72 @@ during a request::
...
+    before returning null. It avoids blocking other receivers from being called.
+
+.. note::
+
+    If the queue name is suffixed by ``.fifo``, AWS will creates a `FIFO queue`_.
```
Comment (javiereguiluz) [inline messenger.rst]: `creates` -> `create`

**Review by javiereguiluz — APPROVED** [review]
Thanks for contributing these docs!

File: messenger.rst (hunk 5 above)
Comment (Nyholm) [reply messenger.rst]: Thank you

File: messenger.rst (hunk 4 above)
Comment (Nyholm) [reply messenger.rst]: Thanks

File: messenger.rst (hunk 3 above)
Comment (Nyholm) [reply messenger.rst]: Thanks

File: messenger.rst (hunk 2 above)
Comment (Nyholm) [reply messenger.rst:1159]: Thank you. 

I leave this unchanged then.

File: messenger.rst [hunk 6]
```diff
@@ -1268,6 +1268,72 @@ during a request::
...
+    The transport will automatically create queues that are needed. This
+    can be disabled setting the "auto_setup" option to ``false``.
+
+A number of options can be configured via the DSN or via the ``options`` key
+under the transport in ``messenger.yaml``:
+
+==================  ======================================  ======================
+     Option         Description                             Default
+==================  ======================================  ======================
+endpoint            Absolute URL to the SQS service         https://sqs.eu-west-1.amazonaws.com
```
Comment (OskarStark) [inline messenger.rst]: Should we sort the option alphabetically?

**Comment by OskarStark** [comment]
Note for the mergers, this must target 5.1 branch

File: messenger.rst (hunk 6 above)
Comment (Nyholm) [reply messenger.rst]: Sure, doing it now

**Comment by Nyholm** [comment]
I've updated the PR

**Comment by wouterj** [comment]
Thanks @Nyholm and @Simperfit!

During the merge, I mistakingly added some commits that should also have made it in 5.1 (from 4.4). So the diff of this PR is a bit weird, but the result is correct.

Afterwards, I've fixed the comment of @javiereguiluz along with some minor formatting changes in 23fd9b401c6b6014551e7ce423d694809b456ecd

**Comment by Nyholm** [comment]
Thank you

**Pull request merged**